[CORE-14704] re-enable post-startup storage checks

dotnwat opened the pull request:
Fixes for observation:

> The code block in redpanda.py (lines 3267-3285) is unreachable because the 'continue' statement is always executed. This is due to a type mismatch: 'NodeStorage' is being checked in a list of 'ClusterNodes', which will always be false. The suggested fix is to add 'node' to 'NodeStorage' and use that in the 'in to_start' check. The storage team is requested to take on this task.

I'm a bit skeptical that this provides much value anymore... the original checks were put in place 5 years ago to verify startup. later it appears they were repurposed as a way to catch issues with cleanup. anyway, they seem cheap. if they give us much issue in the future i think we might want to consider removing them.

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

* none

dotnwat committed:
rptest: accept explicit node list in storage stat collection

This replaces the specific `all_nodes` boolean parameter with an
explicit list of nodes to collect from for expanded flexibility and
updates all call sites.

Signed-off-by: Noah Watkins <noahwatkins@gmail.com>

dotnwat committed:
rptest: reenable pre-flight storage checks

This code was disabled because of a type mismatch issue in which
ClusterNode was being compared with NodeStorage and that is now fixed.
The code itself checks that the Redpanda directory is in an expected
state before starting which may be useful in catching some classes of
failures.

Signed-off-by: Noah Watkins <noahwatkins@gmail.com>

dotnwat committed:
rptest: remove conversion to concrete list

The to_started member is already a set. That it was coverted to a
concrete list is probably a historical artifact given that this code was
all unreachable for quite some time.

Signed-off-by: Noah Watkins <noahwatkins@gmail.com>

dotnwat commented on tests/rptest/services/redpanda.py:
```diff
@@ -5479,32 +5475,35 @@ def node_storage(
         :returns: instances of ClusterStorage
         """
+        if nodes is None:
+            nodes = list(self._started)
```
that tracks

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/77082-->
<!--d16fdc91-2cbf-4544-9744-d47601f1d3a4-->
<!--bfe11054-6041-41c5-b04a-181bd0267dd8-->
<!--d3c002c3-5d38-4dab-addd-09037eb7c6f6-->
<!--f19a01d9-e99a-49b6-85ba-33c5e6092ab6-->
<!--4365757f-05c0-4ecd-b6aa-4b108c657df0-->
<!--063b0859-3065-45ef-beff-9de4ff9873bc-->
<!--afc6840d-07d9-4985-9ab7-c28294ed216e-->
<!--ebaf81e0-ad66-46a1-991d-f07a77b32de0-->
<!--8eca749a-7805-4387-8926-49665300cbb2-->
<!--1fa5d022-c196-4627-870c-5ca602ca5b88-->
<!--b2dac47d-bb0e-4f0d-a54c-61f159c7d624-->
<!--c1741343-4988-4aa7-9d5d-3aa6363ea7c3-->
<!--7b911558-0f66-40ff-812b-d4b7f5905e70-->
<!--3a12c050-26a0-4375-a1e1-3665e4ee0a0f-->
<!--40fd204b-3d42-4592-a157-61d2a64e8557-->
<!--9f1a52d0-f6fd-478c-b3f0-b8126414e070-->
<!--50d61601-242f-4de6-a378-190b615e7fa8-->
<!--551a54ad-7ae5-4776-bd77-6dd7cdecca52-->
<!--b8cf6b60-4bd2-4163-a71a-3b5c4cc29bdf-->
<!--title-->
## Retry command for [Build#77082](https://buildkite.com/redpanda/redpanda/builds/77082)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
skip-redpanda-build
skip-units
skip-rebase
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_no_auth_enabled@{"audit_transport_mode":"rpc"}
tests/rptest/tests/data_transforms_test.py::DataTransformsLoggingTest.test_logs_volume
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_from_end
tests/rptest/tests/audit_log_test.py::AuditLogTestOauth.test_kafka_oauth@{"audit_transport_mode":"rpc","authz_match":"rbac"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_junk_off@{"offset":"__current_time__"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_tracked_offsets_cleaned_up
tests/rptest/tests/audit_log_test.py::AuditLogTestAdminApi.test_audit_log_functioning@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaAuthnApi.test_no_ephemeral_user
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistry.test_sr_audit_bad_authz@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"DELETE_SUBJECT_VERSION"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_SCHEMAS_IDS_ID_SUBJECTS"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_SUBJECT_VERSIONS_VERSION_SCHEMA"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"PUT_MODE"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_public@{"audit_transport_mode":"rpc","endpoint_name":"SCHEMA_REGISTRY_STATUS_READY"}
tests/rptest/tests/follower_fetching_test.py::FollowerFetchingTest.test_basic_follower_fetching@{"fetch_from":"fetch-from-tiered-storage"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaAuthnApi.test_no_audit_user_authn
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistry.test_sr_audit_bad_authz@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestOauth.test_kafka_oauth@{"audit_transport_mode":"kclient","authz_match":"rbac"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"DELETE_SUBJECT_VERSION"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_SCHEMAS_IDS_ID_SUBJECTS"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_SUBJECT_VERSIONS_VERSION_SCHEMA"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"PUT_MODE"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_public@{"audit_transport_mode":"kclient","endpoint_name":"SCHEMA_REGISTRY_STATUS_READY"}
tests/rptest/tests/audit_log_test.py::AuditLogUpgradeTest.test_audit_log_upgrade_all_nodes
tests/rptest/tests/follower_fetching_test.py::FollowerFetchingTest.test_basic_follower_fetching@{"fetch_from":"fetch-from-local"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_no_auth_enabled@{"audit_transport_mode":"kclient"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_compression@{"compression_type":"zstd"}
tests/rptest/tests/data_transforms_test.py::DataTransformsLoggingTest.test_logs_otel
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_junk_off@{"offset":"+NaN"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_patch@{"use_rpk":true}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_compression@{"compression_type":"snappy"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_management@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaAuthnApi.test_excluded_principal@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistry.test_sr_audit_bad_authn@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"DELETE_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_MODE_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_SUBJECT_VERSIONS_VERSION_REFERENCED_BY"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"PUT_CONFIG_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_public@{"audit_transport_mode":"rpc","endpoint_name":"GET_SCHEMAS_TYPES"}
tests/rptest/tests/audit_log_test.py::AuditLogTestsAppLifecycle.test_recovery_mode@{"audit_transport_mode":"rpc"}
tests/rptest/tests/cluster_bootstrap_test.py::ClusterBootstrapFiveNodes.test_topic_creation_during_bootstrap
tests/rptest/tests/audit_log_test.py::AuditLogTestOauth.test_kafka_oauth@{"audit_transport_mode":"rpc","authz_match":"acl"}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"delete","workload":"IDEMPOTENT"}
tests/rptest/tests/follower_fetching_test.py::FollowerFetchingTest.test_basic_follower_fetching@{"fetch_from":"fetch-from-cloud-topic"}
tests/rptest/tests/data_transforms_test.py::DataTransformsLoggingTest.test_log_topic_integrity
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_junk_off@{"offset":"@9223372036854775807"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_patch@{"use_rpk":false}
tests/rptest/tests/topic_creation_test.py::RapidTopicRecreateTest.test_topic_rapid_recreation
tests/rptest/tests/audit_log_test.py::AuditLogTestAdminApi.test_config_rejected
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_compression@{"compression_type":"gzip"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_from_offset@{"offset":"__current_time__"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_identity@{"transactional":false,"wait_running":false}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"delete","workload":"ACKS_1"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_consume@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaAuthnApi.test_authn_failure_messages@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSanctionMode.test_sanctioning_mode@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"DELETE_CONFIG_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_CONFIG_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_SUBJECT_VERSIONS"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"POST_SUBJECT_VERSIONS"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz_get_schemas_ids_id@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestsAppLifecycle.test_app_lifecycle@{"audit_transport_mode":"rpc"}
tests/rptest/tests/cloud_topics/e2e_test.py::EndToEndCloudTopicsTest.test_write
tests/rptest/tests/follower_fetching_test.py::FollowerFetchingTest.test_with_leadership_transfers@{"fetch_from":"fetch-from-cloud-topic"}
tests/rptest/tests/data_transforms_test.py::DataTransformsLoggingMetricsTest.test_logger_metrics_present
tests/rptest/tests/audit_log_test.py::AuditLogTestSmallBuffers.test_bypass@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestAdminApi.test_audit_log_functioning@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_produce@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaTlsApi.test_mtls@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestReproducer.test_sanctioning_mode@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"COMPATIBILITY_SUBJECT_VERSION"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_CONFIG"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_SCHEMAS_IDS_ID_VERSIONS"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"POST_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"PUT_MODE_SUBJECT"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_from_offset@{"offset":"-1"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_off_end@{"offset":"-0"}
tests/rptest/tests/follower_fetching_test.py::FollowerFetchingTest.test_follower_fetching_with_maintenance_mode@{"fetch_from":"fetch-from-cloud-topic"}
tests/rptest/tests/data_transforms_test.py::DataTransformsLoggingTest.test_tunable_configs
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_validation
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_management@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaAuthnApi.test_excluded_principal@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistry.test_sr_audit_bad_authn@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"DELETE_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_MODE_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_SUBJECT_VERSIONS_VERSION_REFERENCED_BY"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"PUT_CONFIG_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_public@{"audit_transport_mode":"kclient","endpoint_name":"GET_SCHEMAS_TYPES"}
tests/rptest/tests/audit_log_test.py::AuditLogTestsAppLifecycle.test_recovery_mode@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestOauth.test_kafka_oauth@{"audit_transport_mode":"kclient","authz_match":"acl"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_compression@{"compression_type":"producer"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_identity@{"transactional":true,"wait_running":true}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"compact","workload":"IDEMPOTENT"}
tests/rptest/tests/partition_movement_test.py::SIPartitionMovementTest.test_cross_shard@{"cloud_storage_type":1,"num_to_upgrade":0,"with_cloud_topics":true}
tests/rptest/tests/data_transforms_test.py::DataTransformsLoggingMetricsTest.test_manager_metrics_values
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_junk_off@{"offset":"+-10"}
tests/rptest/transactions/tx_verifier_test.py::TxVerifierTest.test_all_tx_tests@{"cloud_storage_type":1}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_compression@{"compression_type":"none"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_identity@{"transactional":true,"wait_running":false}
tests/rptest/tests/audit_log_test.py::AuditLogTestOauth.test_admin_oauth@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestAdminAuthApi.test_excluded_principal@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_excluded_topic
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaAuthnApi.test_authn_messages@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistry.test_sr_audit@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"DELETE_MODE_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_MODE"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_SUBJECT_VERSIONS_VERSION"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"PUT_CONFIG"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz_get_subjects@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestsAppLifecycle.test_drain_on_audit_disabled@{"audit_transport_mode":"rpc"}
tests/rptest/tests/follower_fetching_test.py::FollowerFetchingTest.test_with_leadership_transfers@{"fetch_from":"fetch-from-tiered-storage"}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"delete","workload":"ACKS_ALL"}
tests/rptest/tests/data_transforms_test.py::DataTransformsLoggingMetricsTest.test_manager_metrics_present
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_junk_off@{"offset":"-9223372036854775808"}
tests/rptest/transactions/tx_verifier_test.py::TxVerifierTest.test_all_tx_tests@{"cloud_storage_type":2}
tests/rptest/tests/audit_log_test.py::AuditLogTestAdminApi.test_audit_log_metrics@{"audit_transport_mode":"kclient"}
tests/rptest/tests/data_transforms_test.py::DataTransformsMultipleOutputTopicsTest.test_multiple_output_topics
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_cloud_topic_recreation_while_producing
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_produce@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestReproducer.test_sanctioning_mode@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaTlsApi.test_mtls@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"COMPATIBILITY_SUBJECT_VERSION"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_CONFIG"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"GET_SCHEMAS_IDS_ID_VERSIONS"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"POST_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"rpc","endpoint_name":"PUT_MODE_SUBJECT"}
tests/rptest/tests/follower_fetching_test.py::FollowerFetchingTest.test_follower_fetching_with_maintenance_mode@{"fetch_from":"fetch-from-local"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSmallBuffers.test_bypass@{"audit_transport_mode":"rpc"}
tests/rptest/tests/data_transforms_test.py::DataTransformsChainingTest.test_multiple_transforms_chained_together
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_from_offset@{"offset":"+0"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_off_end@{"offset":"@33276193569000"}
tests/rptest/tests/security_report_test.py::AuditlogClientSecurityReportTest.test_security_report
tests/rptest/tests/audit_log_test.py::AuditLogTestAdminApi.test_audit_log_metrics@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_audit_topic_protections
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaAuthnApi.test_authn_failure_messages@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSanctionMode.test_sanctioning_mode@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"DELETE_CONFIG_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_CONFIG_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_SUBJECT_VERSIONS"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"POST_SUBJECT_VERSIONS"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz_get_schemas_ids_id@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestsAppLifecycle.test_app_lifecycle@{"audit_transport_mode":"kclient"}
tests/rptest/tests/cloud_topics/e2e_test.py::EndToEndCloudTopicsTest.test_delete_records
tests/rptest/tests/follower_fetching_test.py::FollowerFetchingTest.test_follower_fetching_with_maintenance_mode@{"fetch_from":"fetch-from-tiered-storage"}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"compact","workload":"ACKS_1"}
tests/rptest/tests/data_transforms_test.py::DataTransformsLeadershipChangingTest.test_leadership_changing_randomly
tests/rptest/tests/data_transforms_test.py::DataTransformsRpcUpgradeTest.test_upgrade_one_node
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_from_offset@{"offset":"+9223372036854775807"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_off_end@{"offset":null}
tests/rptest/tests/partition_movement_test.py::SIPartitionMovementTest.test_shadow_indexing@{"cloud_storage_type":2,"num_to_upgrade":0,"with_cloud_topics":true}
tests/rptest/tests/audit_log_test.py::AuditLogTestOauth.test_admin_oauth@{"audit_transport_mode":"kclient"}
tests/rptest/tests/cloud_topics/e2e_test.py::EndToEndCloudTopicsTxTest.test_write
tests/rptest/tests/audit_log_test.py::AuditLogTestAdminAuthApi.test_excluded_principal@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaApi.test_consume@{"audit_transport_mode":"rpc"}
tests/rptest/tests/audit_log_test.py::AuditLogTestKafkaAuthnApi.test_authn_messages@{"audit_transport_mode":"kclient"}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"compact","workload":"ACKS_ALL"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistry.test_sr_audit@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"DELETE_MODE_SUBJECT"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_MODE"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"GET_SUBJECT_VERSIONS_VERSION"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz@{"audit_transport_mode":"kclient","endpoint_name":"PUT_CONFIG"}
tests/rptest/tests/audit_log_test.py::AuditLogTestSchemaRegistryACLs.test_sr_audit_authz_get_subjects@{"audit_transport_mode":"kclient"}
tests/rptest/tests/audit_log_test.py::AuditLogTestsAppLifecycle.test_drain_on_audit_disabled@{"audit_transport_mode":"kclient"}
tests/rptest/tests/follower_fetching_test.py::FollowerFetchingTest.test_with_leadership_transfers@{"fetch_from":"fetch-from-local"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_compression@{"compression_type":"lz4"}
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_identity@{"transactional":false,"wait_running":true}
tests/rptest/tests/data_transforms_test.py::DataTransformsLoggingMetricsTest.test_logger_metrics_values
tests/rptest/tests/data_transforms_test.py::DataTransformsTest.test_consume_junk_off@{"offset":"--10"}
```
<!--/body-->

dotnwat committed:
tests: update post-start check with newer state manifest

A lot has changed since this check was added and later rendered
unreachable. So update it with the latest bits.

Signed-off-by: Noah Watkins <noahwatkins@gmail.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/77092-->
<!--f5d49d64-0759-451b-b0b7-b5d6f6cf479e-->
<!--0d6fe120-3bb1-4e97-9c09-aff42fdb0af4-->
<!--title-->
## Retry command for [Build#77092](https://buildkite.com/redpanda/redpanda/builds/77092)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
skip-redpanda-build
skip-units
skip-rebase
tests/rptest/tests/cluster_bootstrap_test.py::ClusterBootstrapFiveNodes.test_topic_creation_during_bootstrap
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--e2b3bb42-d01e-4f38-a9ad-f083814bfaff-->
<!--TEST RESULTS-->
<!--779310cb-795f-4452-a5d6-908f46e3a987-->
<!--TEST RESULTS-->
<!--906ee3c7-c2c1-4936-91ab-a03f956be4cf-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/77092">build#77092</a></summary>


| test_class                | test_method                          | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 | test_history                                                                                                                                                                 |
|:--------------------------|:-------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ClusterBootstrapFiveNodes | test_topic_creation_during_bootstrap | null             | integration | https://buildkite.com/redpanda/redpanda/builds/77092#019ac7a6-c5dc-498b-a532-58eff8cb41a7 | FAIL          | 0/21     | The test has failed across all retries | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterBootstrapFiveNodes&test_method=test_topic_creation_during_bootstrap |
| ClusterBootstrapFiveNodes | test_topic_creation_during_bootstrap | null             | integration | https://buildkite.com/redpanda/redpanda/builds/77092#019ac7ad-dbe9-435a-a278-1dd05a3f910d | FAIL          | 0/21     | The test has failed across all retries | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterBootstrapFiveNodes&test_method=test_topic_creation_during_bootstrap |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/77106">build#77106</a></summary>


| test_class                | test_method                          | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 | test_history                                                                                                                                                                 |
|:--------------------------|:-------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ClusterBootstrapFiveNodes | test_topic_creation_during_bootstrap | null             | integration | https://buildkite.com/redpanda/redpanda/builds/77106#019acb1e-e984-4bb1-9daa-aec92d9795d8 | FAIL          | 0/21     | The test has failed across all retries | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterBootstrapFiveNodes&test_method=test_topic_creation_during_bootstrap |
| ClusterBootstrapFiveNodes | test_topic_creation_during_bootstrap | null             | integration | https://buildkite.com/redpanda/redpanda/builds/77106#019acb24-ff12-4f2e-bfa9-8716da96f5c4 | FAIL          | 0/21     | The test has failed across all retries | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterBootstrapFiveNodes&test_method=test_topic_creation_during_bootstrap |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/77107">build#77107</a></summary>


| test_class              | test_method         | test_arguments            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                              |
|:------------------------|:--------------------|:--------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|
| MountUnmountIcebergTest | test_simple_remount | {"cloud_storage_type": 1} | integration | https://buildkite.com/redpanda/redpanda/builds/77107#019acb91-7924-4970-85a0-f9665152af7b | FLAKY         | 14/21    | upstream reliability is '79.95735607675905'. current run reliability is '66.66666666666666'. drift is 13.29069 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=MountUnmountIcebergTest&test_method=test_simple_remount |
</details>
<!--/body-->

dotnwat committed:
rptest: skip storage init checks in test

The ClusterBootstrapFiveNodes/test_topic_creation_during_bootstrap test
explicitly runs operations, whose side effects include creation of
consumer offsets topic and test topic, concurrently with cluster startup
which means that immediately after bootup a broker won't have a
predictable storage layout that is non-dependent on the test itself.
Therefore we just disable the checks on this one test. IF we needed to
do this a lot we could create a simple protocol for tests to express
customizations to the check.

Signed-off-by: Noah Watkins <noahwatkins@gmail.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/77106-->
<!--a62df10f-1b69-45c4-965c-f13a282c0ecf-->
<!--7bae9c7e-272a-4896-8f37-309bf3616db3-->
<!--title-->
## Retry command for [Build#77106](https://buildkite.com/redpanda/redpanda/builds/77106)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
skip-redpanda-build
skip-units
skip-rebase
tests/rptest/tests/cluster_bootstrap_test.py::ClusterBootstrapFiveNodes.test_topic_creation_during_bootstrap
```
<!--/body-->

nvartolomei commented on tests/rptest/services/redpanda.py:
```diff
@@ -3374,9 +3374,17 @@ def start_one(node: ClusterNode):
+            "kafka": {"_redpanda.audit_log", "_redpanda.transform_logs"},
+            "kafka_internal": {"ct_l1_domain"},
+        }
+        expected["l1_staging"] = set()  # make type deduction happy
```
Don't get this line.

dotnwat replied:
pyright doesn't like it if you initialize with an empty set in the above construction because its trying to deduce the type and can't. here it also can't (i think), but it doesn't complain. maybe there is a syntax for it?
